Content.Server/GameObjects/EntitySystems/HandsSystem.cs

```diff
@@ -190,5 +199,56 @@ private bool HandleThrowItem(ICommonSession session, GridCoordinates coords, Ent
 
             return true;
         }
+
+        private bool HandleSmartEquipBackpack(ICommonSession session, GridCoordinates coords, EntityUid uid)
+        {
+            return HandleSmartEquip(session, coords, EquipmentSlotDefines.Slots.BACKPACK);
+        }
+
+        private bool HandleSmartEquipBelt(ICommonSession session, GridCoordinates coords, EntityUid uid)
+        {
+            return HandleSmartEquip(session, coords, EquipmentSlotDefines.Slots.BELT);
+        }
+
+        private bool HandleSmartEquip(ICommonSession session, GridCoordinates coords, EquipmentSlotDefines.Slots equipementSlot)
+        {
+            var plyEnt = ((IPlayerSession) session).AttachedEntity;
+
+            if (plyEnt == null || !plyEnt.IsValid())
+                return false;
+
+            if (!plyEnt.TryGetComponent(out HandsComponent handsComp) || !plyEnt.TryGetComponent(out InventoryComponent inventoryComp))
+                return false;
+
+            if (!inventoryComp.TryGetSlotItem(equipementSlot, out ItemComponent equipmentItem)
+                || !equipmentItem.Owner.TryGetComponent<ServerStorageComponent>(out var storageComponent))
+            {
+                _notifyManager.PopupMessage(plyEnt, plyEnt, Loc.GetString("You have no {0} to take something out of!", EquipmentSlotDefines.SlotNames[equipementSlot].ToLower()));
+                return true;
+            }
+
+            var heldItem = handsComp.GetHand(handsComp.ActiveIndex)?.Owner;
+
+            if (heldItem != null)
+            {
+                var interactionSystem = _entitySystemManager.GetEntitySystem<InteractionSystem>();
+                interactionSystem.Interaction(plyEnt, heldItem, equipmentItem.Owner, coords);
```

Comment:
This should directly tell the storage component to insert the item, not round trip through the interaction system.

Reply:
Also, passing `coords` here is incorrect because those coords aren't actually related to the object, they're just where the user's mouse happened to be.

As an extension of that, you can stop using a `PointerInputCmdHandler` and instead use `InputCmdHandler.FromDelegate`, since you don't need coord/entity information from the input event itself.

Reply:
Done. I made the method `ServerStorageComponent.PlayerInsertEntity` public for this.